apps/next-app/src/app/products/page.tsx

```diff
@@ -182,63 +228,94 @@ export default function ProductsPage() {
 
   return (
     <>
-      <button
+      <Button
         onClick={() => {
           router.push('/');
         }}
       >
         Homepage
-      </button>
-      <button
-        onClick={() => {
-          router.push('/products/product1');
-        }}
-      >
-        DetailProduct
-      </button>
+      </Button>
       <div className='bg-neutral-100'>
-        <div className='max-w-7xl mx-auto w-full '>
-          <div className='grid grid-cols-12 gap-3 md:!gap-7'>
-            <div className='col-span-12 md:col-span-3'></div>
-            <div className='col-span-12 md:col-span-9 px-[5px] md:px-0 py-[10px] md:py-[30px] flex items-center font-primary'>
+        <div className='max-w-7xl mx-auto w-full'>
+          <div className='bg-white md:!bg-neutral-100 grid grid-cols-12 mb-1 mt-2'>
+            <div className='col-span-12 md:col-span-3 md:!gap-x-7'></div>
+            <div className='col-span-12 md:col-span-9 px-[5px] md:px-0 py-[10px] md:!py-[30px] flex items-center font-primary'>
               {/*#0F0F0F Onyx color not found*/}
+              {/* md:text-primary doesn't work */}
               <span className='text-[#0F0F0F] md:text-primary text-base md:text-2xl font-semibold md:font-bold mr-[5px]'>
                 {title.toUpperCase()}
               </span>
               {/*#575757 Onyx color not found*/}
               <span className='text-[#575757] md:text-black text-[11px] md:text-base font-normal'>{`(${quantity} sản phẩm)`}</span>
             </div>
+          </div>
+          <div className='grid grid-cols-12 gap-3 md:!gap-x-7'>
             <div className='col-span-12 md:!col-span-3'>
-              <FilterComponent
-                filterGroupItems={filterData}
-                applyFilterTitle={'Áp dụng'}
-                clearFilterTitle={'Xóa bộ lọc'}
-                filterTitle={'Bộ Lọc'}
-                onClickSortAndFilter={onClickSortAndFilter}
-                onChangePageLayout={onChangePageLayout}
-              />
+              <div className='hidden md:block'>
+                {filterData.sortItems ? (
+                  <div key={filterData.sortItems.filterId} className='mb-12'>
+                    <FilterItem
+                      filterItems={filterData.sortItems}
+                      onClickSort={onClickSort}
+                    ></FilterItem>
+                  </div>
+                ) : (
+                  <></>
+                )}
```

Comment:
This component design looks pretty bad IMO. This Filter component should be self-contained/ reusable, and manage it own selected state. 
```
const selectedSortId = searchParams.get('sort')
```
```suggestion
                {filterData.sortItems && (
                  <FilterSelect selected={selectedSortId } options={filterData.sortItems}>
                )}
```
Using this approach, we only need one `FilterSelect` and reuse for other filter type

Reply:
Well, first, component shouldn't handle null value in it, if you mean it can handle selected value or default value then you can config by value from filterItems -> selected. 
When we work from sever site, data must correct, and it already have default value. In this case, value is handled by server, not it's self. Besides that, client site can like you recommend because data don't need submit to server atm so you need field to handle by it's self.